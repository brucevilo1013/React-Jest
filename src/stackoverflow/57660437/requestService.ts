export async function runningSince() {
  return {
    data: {
      runningSince: 'real data'
    }
  };
}
