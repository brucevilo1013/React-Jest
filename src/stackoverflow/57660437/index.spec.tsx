import React from 'react';
import { Header } from './';
import { shallow } from 'enzyme';
import * as requestService from './requestService';

jest.mock('./requestService.ts', () => {
  return {
    runningSince: jest.fn()
  };
});

describe('Header', () => {
  it('t1', done => {
    const mockedData = {
      data: {
        runningSince: new Date('2019-08-26T11:55:03.696Z').toISOString()
      }
    };
    (requestService.runningSince as jest.MockedFunction<typeof requestService.runningSince>).mockResolvedValueOnce(
      mockedData
    );
    const wrapper = shallow(<Header></Header>);
    setImmediate(() => {
      expect(wrapper.state()).toEqual({ startDate: mockedData.data.runningSince });
      expect(wrapper.text()).toEqual(mockedData.data.runningSince);
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
