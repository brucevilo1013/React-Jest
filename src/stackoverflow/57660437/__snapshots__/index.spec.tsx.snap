// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header t1 1`] = `
<div>
  2019-08-26T11:55:03.696Z
</div>
`;
